src/main/java/net/dries007/tfc/compat/jade/common/BlockEntityTooltips.java

```diff
@@ -0,0 +1,332 @@
+import net.minecraft.world.level.Level;
+import net.minecraft.world.level.block.entity.BlockEntity;
+import net.minecraft.world.level.block.state.BlockState;
+
+import net.dries007.tfc.common.blockentities.*;
+import net.dries007.tfc.common.blocks.crop.CropBlock;
+import net.dries007.tfc.common.blocks.devices.*;
+import net.dries007.tfc.common.blocks.plant.fruit.FruitTreeSaplingBlock;
+import net.dries007.tfc.common.blocks.soil.HoeOverlayBlock;
+import net.dries007.tfc.common.blocks.wood.TFCSaplingBlock;
+import net.dries007.tfc.common.capabilities.Capabilities;
+import net.dries007.tfc.common.capabilities.egg.EggCapability;
+import net.dries007.tfc.common.capabilities.egg.IEgg;
+import net.dries007.tfc.common.capabilities.food.FoodCapability;
+import net.dries007.tfc.common.capabilities.heat.HeatCapability;
+import net.dries007.tfc.common.recipes.BarrelRecipe;
+import net.dries007.tfc.common.recipes.BloomeryRecipe;
+import net.dries007.tfc.common.recipes.SoupPotRecipe;
+import net.dries007.tfc.config.TFCConfig;
+import net.dries007.tfc.util.Helpers;
+import net.dries007.tfc.util.LampFuel;
+import net.dries007.tfc.util.calendar.Calendars;
+import net.dries007.tfc.util.calendar.ICalendar;
+
+/**
+ * Common tooltips that can be displayed for various block entities via external sources.
+ */
+public final class BlockEntityTooltips
+{
+    public static final BlockEntityTooltip BARREL = (level, state, entity, tooltip) -> {
+        if (state.getBlock() instanceof BarrelBlock && entity instanceof BarrelBlockEntity barrel)
+        {
+            if (state.getValue(BarrelBlock.SEALED))
+            {
+                BarrelRecipe recipe = barrel.getRecipe();
+                if (recipe != null)
+                {
+                    tooltip.accept(recipe.getTranslationComponent());
+                    tooltip.accept(Helpers.translatable("tfc.jade.sealed_date").append(ICalendar.getTimeAndDate(Calendars.CLIENT.ticksToCalendarTicks(barrel.getSealedTick()), Calendars.get(level).getCalendarDaysInMonth())));
+                }
+            }
+        }
+    };
+
+    public static final BlockEntityTooltip BELLOWS = (level, state, entity, tooltip) -> {
+        if (entity instanceof BellowsBlockEntity bellows)
+        {
+            int pushTicks = bellows.getTicksSincePushed();
+            if (pushTicks < 20)
+            {
+                pushTicks *= 2;
+                if (pushTicks > 20)
+                {
+                    pushTicks = 40 - pushTicks;
+                }
+
+                MutableComponent component = Helpers.translatable("tfc.jade.bellows_start");
+                String middle = Helpers.translatable("tfc.jade.bellows_middle").getString(); // probably better to translate this once.
+                for (int i = 0; i < pushTicks; i++)
+                {
+                    component.append(Helpers.literal(middle));
+                }
+                component.append(Helpers.translatable("tfc.jade.bellows_end"));
```

Comment:
As discussed, replacing this with explicit translations would be better. I would suggest divide `pushTicks` by two, then have between `Woosh` and Woooooooooooosh`.